src/Loader/SizeCheckingLoader.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Tuf\Loader;
+
+use Psr\Http\Message\StreamInterface;
+use Tuf\Exception\DownloadSizeException;
+
+/**
+ * A data loader that enforces a size limit on the output of another loader.
+ */
+class SizeCheckingLoader implements LoaderInterface
+{
+    public function __construct(private LoaderInterface $decorated)
+    {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function load(string $uri, int $maxBytes = null): StreamInterface
+    {
+        $data = $this->decorated->load($uri, $maxBytes);
+
+        if (isset($maxBytes)) {
```

Comment:
It seems like that fact that this could be NULL defeats the whole purpose. 

Could we enforce it being not NULL?

Reply:
After discussion on Zoom, maxBytes is now required by the interface.